theano/gpuarray/opt.py

```diff
@@ -1075,15 +1074,14 @@ def local_gpua_careduce(op, context_name, inputs, outputs):
             # be None
             reshaped_gpu_inputs = [gpu_reshaped_x]
             if greduce.supports_c_code(reshaped_gpu_inputs):
-                reduce_reshaped_x = host_from_gpu(
-                    greduce(gpu_reshaped_x))
+                reduce_reshaped_x = greduce(gpu_reshaped_x)
 
                 if reduce_reshaped_x.ndim != outputs[0].ndim:
                     out_shp = []
                     for i in range(x.ndim):
                         if i not in op.axis:
                             out_shp.append(shape_i(x, i))
-                    unreshaped_reduce = reduce_reshaped_x.reshape(
+                    unreshaped_reduce = GpuReshape(len(out_shp))(
```

Comment:
@nouiz  This cause tests to fail. I had tested the theano/gpuarray/tests/test_opt.py but didn't give me any failures or errors.
Shouldn't that line be something like this , 
`unreshaped_reduce = GpuReshape(len(out_shp), None)(*tensor.stack(out_shp))`
I tried that and I get this error, 
```
ERROR (theano.gof.opt): Optimization failure due to: local_gpua_careduce
ERROR (theano.gof.opt): node: Sum{axis=[0], acc_dtype=float64}(HostFromGpu(gpuarray).0)
ERROR (theano.gof.opt): TRACEBACK:
ERROR (theano.gof.opt): Traceback (most recent call last):
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 1820, in process_node
    replacements = lopt.transform(node)
  File "/Users/Ramana/projects/theano/Theano/theano/gpuarray/opt.py", line 196, in local_opt
    new_op = maker(node.op, context_name, node.inputs, node.outputs)
  File "/Users/Ramana/projects/theano/Theano/theano/gpuarray/opt.py", line 1093, in local_gpua_careduce
    *tensor.stack(out_shp))
  File "/Users/Ramana/projects/theano/Theano/theano/gof/op.py", line 602, in __call__
    node = self.make_node(*inputs, **kwargs)
TypeError: make_node() takes exactly 3 arguments (4 given)
```